text/0000-posix-error-numbers.md

```diff
@@ -0,0 +1,248 @@
+- Feature Name: posix_error_numbers
+- Start Date: 2020-08-09
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Add a `std::os::unix::ErrorNumber` struct for storing and comparing POSIX error
+numbers.
+
+# Motivation
+[motivation]: #motivation
+
+The POSIX standard defines a set of named [error numbers], such as `EIO` or
+`EPIPE`, for use in communicating error conditions from low-level OS routines.
+Many of these are currently exposed in [`libc`] as `c_int` constants, because
+they figure prominently in writing bindings to libc on UNIX platforms.
+
+However, there are two problems with this solution:
+
+* POSIX does not specify the type of these error numbers -- they are defined as
+  C macros, which are untyped integer literals. Many traditional UNIX APIs
+  assume that they can use any integer type for returning error numbers, which
+  are unergonomic to compare with a `c_int` const.
+
+* Binaries that interact with the kernel directly (e.g. via `asm!()` syscalls)
+  need to interpret error numbers, and should not require a dependency on libc
+  to do so.
+
+I propose adding an opaque `ErrorNumber` struct to `std::os::unix`, which would
```

Comment:
1. My understanding of `core` is that it is for platform-independent code. Since error numbers are OS-specific, they should be placed in `std` and also be available through a separate sub-crate, similar to the `alloc`.
2. The set of error numbers would be difficult to express and use as a `enum` due to size and the presence of duplicate integer values for some symbolic names.

Reply:
lgtm. maybe insert a small code example?